### DISCUSSION TITLE
Get a New Page whilst in Enum Block

### DISCUSSION THREAD

**Discussion by HxTrinity** [typst/typst#3707] [Q&A] [answered]
Hello, I'm wondering whether it is possible to go onto a new page with enum instead of adding a lot of line breaks till it reaches the new page.

For example:

```typst
#enum(numbering: it => strong[#numbering("Q5.1)", it)])[
    This is the first question
][
    This is the next question, oh no I am reaching the bottom of the page. Better to start this on a new page. \ \ \ \ \ \ \ \ \
     
    Is there a way of getting new page without a load of line breaks whilst in a enum block?
]

```

I've tried `#pagebreak()` but I don't think I can do that whilst in the enum block and the way I've been doing this is by spamming a load of `\` and trying to get just enough so it starts on the next page.

Any help is appreciated!

Thanks!

**Comment by emilyyyylime** [comment]
I tend to use `#v(1fr)`

**Reply by HxTrinity** [reply]
Whilst that does work if I decide to move onto the next item in the enum (next question), I want it to be able to go to the next page on the same item (question). 

When I try this it appears at the bottom of the page and not a new page. 


Example:

Q5.1) This is the first question

I want this line to be on a new page but still part of Q5.1

Thanks,

**Comment by HxTrinity** [comment, marked as answer]
Had a further read of the docs.

Found the solution

You can use `#colbreak()` and it starts a new page inside a enum container. No idea how that works but it does :)

**Comment by ludwig-austermann** [comment]
I think you are approaching your problem wrongly. It seems to me, that you only want/need the numbering of the enum. Then however, I would rather define my own command with a counter inside, e.g. as below.

```typst
#let question-counter = counter("question")
#let question(body, sub: false) = {
  if sub {
    question-counter.step(level: 2)
  } else {
    question-counter.step()
  }
  strong(question-counter.display("Q1.1) "))
  body
  pagebreak(weak: true)
}

#question[Do you like this alternative?]
#question(sub: true)[How do you rate the alternative?]
#question(sub: true)[Do you that this way you can better customize everything?]
#question[Do you have other ideas?]
```
